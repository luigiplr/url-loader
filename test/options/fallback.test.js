/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('fallback', () => {
    test('{String}', async () => {
      const config = {
        loader: {
          test: /\.png$/,
          options: {
            limit: 100,
            fallback: 'file-loader',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { source } = stats.toJson().modules[1];

      expect(source).toMatchSnapshot();
    });

    test('{Function}', async () => {
      const config = {
        loader: {
          test: /\.png$/,
          options: {
            limit: 100,
            fallback: function loader() {
              return "module.exports = 'https://www.vectorlogo.zone/logos/js_webpack/js_webpack-card.png'";
            },
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { source } = stats.toJson().modules[1];

      expect(source).toMatchSnapshot();
    });
  });
});
