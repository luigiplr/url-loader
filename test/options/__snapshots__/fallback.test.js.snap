// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options fallback {Function} 1`] = `"throw new Error(\\"Module build failed: Error: schema is invalid: data.properties['fallback'].type should be equal to one of the allowed values, data.properties['fallback'].type[1] should be equal to one of the allowed values, data.properties['fallback'].type should match some schema in anyOf\\\\n    at Ajv.validateSchema (D:\\\\\\\\git\\\\\\\\luigiplr\\\\\\\\url-loader\\\\\\\\node_modules\\\\\\\\ajv\\\\\\\\lib\\\\\\\\ajv.js:188:16)\\\\n    at Ajv._addSchema (D:\\\\\\\\git\\\\\\\\luigiplr\\\\\\\\url-loader\\\\\\\\node_modules\\\\\\\\ajv\\\\\\\\lib\\\\\\\\ajv.js:317:10)\\\\n    at Ajv.validate (D:\\\\\\\\git\\\\\\\\luigiplr\\\\\\\\url-loader\\\\\\\\node_modules\\\\\\\\ajv\\\\\\\\lib\\\\\\\\ajv.js:97:26)\\\\n    at validateOptions (D:\\\\\\\\git\\\\\\\\luigiplr\\\\\\\\url-loader\\\\\\\\node_modules\\\\\\\\schema-utils\\\\\\\\dist\\\\\\\\validateOptions.js:48:12)\\\\n    at Object.loader (D:\\\\\\\\git\\\\\\\\luigiplr\\\\\\\\url-loader\\\\\\\\src\\\\\\\\index.js:22:84)\\");"`;

exports[`Options fallback {String} 1`] = `"throw new Error(\\"Module build failed: Error: schema is invalid: data.properties['fallback'].type should be equal to one of the allowed values, data.properties['fallback'].type[1] should be equal to one of the allowed values, data.properties['fallback'].type should match some schema in anyOf\\\\n    at Ajv.validateSchema (D:\\\\\\\\git\\\\\\\\luigiplr\\\\\\\\url-loader\\\\\\\\node_modules\\\\\\\\ajv\\\\\\\\lib\\\\\\\\ajv.js:188:16)\\\\n    at Ajv._addSchema (D:\\\\\\\\git\\\\\\\\luigiplr\\\\\\\\url-loader\\\\\\\\node_modules\\\\\\\\ajv\\\\\\\\lib\\\\\\\\ajv.js:317:10)\\\\n    at Ajv.validate (D:\\\\\\\\git\\\\\\\\luigiplr\\\\\\\\url-loader\\\\\\\\node_modules\\\\\\\\ajv\\\\\\\\lib\\\\\\\\ajv.js:97:26)\\\\n    at validateOptions (D:\\\\\\\\git\\\\\\\\luigiplr\\\\\\\\url-loader\\\\\\\\node_modules\\\\\\\\schema-utils\\\\\\\\dist\\\\\\\\validateOptions.js:48:12)\\\\n    at Object.loader (D:\\\\\\\\git\\\\\\\\luigiplr\\\\\\\\url-loader\\\\\\\\src\\\\\\\\index.js:22:84)\\");"`;
